sparkplug/timereporters/base.py

```diff
@@ -0,0 +1,43 @@
+import datetime
+
+##################################################
+
+def _milliseconds( timedelta ):
+    return timedelta / datetime.timedelta( microseconds=1000 )
+
+
+##################################################
+
+def mn_md_mx( l ):
+    "calculate min median max from a list of timedeltas, returns the time in milliseconds"
+    mn = 0
+    median = 0
+    mx = 0
+    if l :
+        lenl = len(l)
+        ordered = sorted( l )
+        mn = _milliseconds( ordered[0] )
+        mx = _milliseconds( ordered[-1] )
+        if lenl & 0x01 :
+            # odd number of samples
+            median = _milliseconds( ordered[ lenl >> 1 ] )
+        else:
+            # even number of samples
+            median = 0.5 * _milliseconds( ordered[ lenl >> 1 ] + ordered[ ( lenl >> 1 ) - 1 ])
+    return( mn, median, mx )
```

Comment:
you don't need the brackets here, but if you want to use them, leave a space between `return` the the opening bracket. Kind of makes it look like `return` is a function like this